js/index.js

```diff
@@ -141,12 +141,33 @@ popupClose.addEventListener('click', () => {
 const form = document.querySelector('.form');
 const email = document.getElementById('form-email');
 const error = document.querySelector('.error');
+const text = document.querySelector('#form-in1');
+const mess = document.querySelector('.form-msg');
+
 form.addEventListener('submit', (e) => {
   const emailValue = email.value;
+  let flag = 0;
   for (let i = 0; i < emailValue.length; i += 1) {
     if (emailValue[i] !== emailValue[i].toLowerCase()) {
       e.preventDefault();
       error.innerText = 'Please enter email in lowercase';
+      flag = 1;
     }
   }
+
+  if (flag === 0) {
+    const formData = {
+      name: text.value,
+      email: email.value,
+      message: mess.value,
+    };
+    localStorage.setItem('data', JSON.stringify(formData));
+  }
```

Comment:
**OPTIONAL** - According to the project requirements, "After filling the form, and not submitting it, the entered values should be saved in the browser." Currently, your app doesn't function this way, I advise you to fix this issue 🙏. You can use the `input` event listener on your form to implement this functionality.

Your project:
- As you'd notice, I have filled the form, but the data is not preserved in the localstorage.
<img width="960" alt="bndbjndb" src="https://user-images.githubusercontent.com/73607512/202670702-5596f25a-b458-468f-b9c4-4dbd07584821.PNG">